Code clean up and folder restructuring

VladIulian96 opened the pull request.

VladIulian96 committed:
Code clean up and folder restructuring

VladIulian96 committed:
Added a validation when creating a new product

VladIulian96 committed:
Merge branch 'refs/heads/master' into products-controller

adrianbucur83 commented on src/main/java/com/example/demo1/controller/ProductsController.java:
```diff
@@ -16,48 +17,42 @@ public class ProductsController {
     public String edit_product_form(@PathVariable(name = "id") int productId, Model model) {
-        model.addAttribute("product", productService.getProductById(productId));
-        return "updateProductPage";
+        if(productService.productExists(productId)) {
```
productExists most likely does findById so you're doing the same work twice

adrianbucur83 commented on src/main/java/com/example/demo1/controller/ProductsController.java:
```diff
@@ -16,48 +17,42 @@ public class ProductsController {
-//            oldProduct.setDescription(newProduct.getDescription());
-
-            productService.updateProduct(oldProduct, newProduct);
+            productsRepository.save(newProduct);
```
add validation and a service where you work with the database and do processing. don't call the db in the controller. Also use DTOs to get and return data

adrianbucur83 commented on src/main/java/com/example/demo1/controller/ProductsController.java:
```diff
@@ -73,29 +68,23 @@ public String createProduct(HttpServletRequest request, RedirectAttributes redir
             redirectAttributes.addFlashAttribute("createMessage", "Product created!");
             productService.createProduct(product);
         } else {
             redirectAttributes.addFlashAttribute("createMessage", "A product with this id already exists!");
```
why do you need flash attribute for a string?

adrianbucur83 commented on src/main/java/com/example/demo1/service/impl/ProductService.java:
```diff
@@ -1,28 +1,24 @@
-    public List<Product> getAllProducts() {
-        return new ArrayList<>(productHashMap.values());
+    public ProductService(ProductsRepository productsRepository) {
+        this.productsRepository = productsRepository;
```
you can use lombok

adrianbucur83 commented on src/main/java/com/example/demo1/service/impl/ProductService.java:
```diff
@@ -1,28 +1,24 @@
 
     public void createProduct(Product product) {
-        productHashMap.put(product.getId(), product);
+        if(!productExists(product.getId())) {
```
this logic should be in a service

adrianbucur83 replied:
also: ProductCreateDto

adrianbucur83 reviewed (changes requested):
Use dtos, create the service class, validate dtos for create and update

VladIulian96 committed:
- created ProductDto.java .
- recoded the updateProduct method to use the ProductDto.
- Now, the createProduct method checks if there is already a product with the same name instead of checking if the ID is the same.
- Added client side validation to some fields
